libcontainer/cgroups/systemd/common.go

```diff
@@ -231,18 +230,21 @@ func systemdVersion(cm *dbusConnManager) int {
 	return version
 }
 
-func systemdVersionAtoi(verStr string) (int, error) {
-	// verStr should be of the form:
-	// "v245.4-1.fc32", "245", "v245-1.fc32", "245-1.fc32" (without quotes).
+func systemdVersionAtoi(str string) (int, error) {
+	// str should be of the form:
+	// "v245.4-1.fc32", "245", "v245-1.fc32", "245-1.fc32" (with quotes).
 	// The result for all of the above should be 245.
-	// Thus, we unconditionally remove the "v" prefix
+	// Thus, we unconditionally remove the "v prefix
 	// and then match on the first integer we can grab.
-	re := regexp.MustCompile(`v?([0-9]+)`)
-	matches := re.FindStringSubmatch(verStr)
-	if len(matches) < 2 {
-		return 0, fmt.Errorf("can't parse version %s: incorrect number of matches %v", verStr, matches)
+	str = strings.TrimLeft(str, `"v`)
+	for i := 0; i < len(str); i++ {
+		if str[i] < '0' || str[i] > '9' {
+			// First non-digit.
```

Comment:
Had my wires crossed initially, and thought this was stripping any _leading_ non-digit characters (so making `"vHELLO245.4-1.fc32"` valid), but because we strip the `"v` first, this look (should) start at `245.4-1.fc32`, and removes the _trailing_ non-digit parts (`.4-1.fc32"`).

Perhaps

```suggestion
			// Strip non-digit suffix and quote (e.g. `.4-1.fc32"`).
```

Reply:
So, this comment is a logical continuation of the previous one ("match on the first integer we can grab"). While that comment above is, I think, sufficient, I am adding a little bit more here.

Changed to `First non-digit: cut the tail.`